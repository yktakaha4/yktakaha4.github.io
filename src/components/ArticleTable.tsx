import dayjs from "dayjs";
import {FC} from "react";

export interface ArticleTableRow {
    title: string
    url: string
    likes: number
    publishedAt: Date
}

export interface ArticleTableProps {
    rows: Array<ArticleTableRow>
}

export const ArticleTable: FC<ArticleTableProps> = ({ rows }) => {
    return (
        <table>
            <thead>
            <tr>
                <th style={{ textAlign: 'center' }}>公開日</th>
                <th>記事名</th>
            </tr>
            </thead>
            <tbody>
            {
                rows.map(row => (
                    <tr>
                        <td>{dayjs(row.publishedAt).format('YYYY/M/D')}</td>
                        <td><a href={row.url} target="_blank" rel="noopener noreferrer">{row.title}</a></td>
                    </tr>
                ))
            }
            </tbody>
        </table>
    )
}
