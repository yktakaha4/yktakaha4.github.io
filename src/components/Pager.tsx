import React, { FC } from 'react';

export interface PagerProps {
    page: number
    perPage: number
    total: number
    size: number
    onChange?: (args: { page: number }) => void
}

export const Pager: FC<PagerProps> = ({ page, perPage, total, size, onChange }) => {
    const minSize = 3
    if (size < minSize) {
        throw new Error(`size must be greater than or equal to ${minSize}`)
    }

    const totalPages = Math.ceil(total / perPage);
    const currentPage = Math.min(Math.max(1, page), totalPages);
    const startPage = Math.min(Math.max(1, currentPage - Math.floor(size / 2)), Math.max(totalPages - size + 1, 1));
    const pagesLength = Math.max(1, Math.min(size, totalPages))
    const pages = Array.from({ length: pagesLength }).map((_, i) => startPage + i);

    const handleClick = (p: number) => {
        return (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
            e.preventDefault();
            if (onChange) {
                onChange({ page: p })
            }
        }
    }

    return (
        <ul className="pagination" style={{margin: 0}}>
            <li className="pagination__item">
                <a className="pagination__link" href="#" onClick={handleClick(1)}>&laquo;</a>
            </li>
            {pages.length > 1 && pages[0] !== 1 && (
                <li className="pagination__item">
                    <span>…</span>
                </li>
            )}
            {pages.map(p => (
                <li key={p} className={`pagination__item ${p === currentPage ? 'pagination__item--active' : ''}`}>
                    <a className="pagination__link" href="#" onClick={handleClick(p)}>{p}</a>
                </li>
            ))}
            {pages.length > 1 && pages[pages.length - 1] !== totalPages && (
                <li className="pagination__item">
                    <span>…</span>
                </li>
            )}
            <li className="pagination__item">
                <a className="pagination__link" href="#" onClick={handleClick(totalPages)}>&raquo;</a>
            </li>
        </ul>
    );
}
