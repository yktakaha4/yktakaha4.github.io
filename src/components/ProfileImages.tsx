import { FC } from 'react';
import { Avatar } from '@/components/ui/Avatar';

export const ProfileImages: FC = () => {
  const src = [
    //
    require('@site/static/img/icon.jpg').default,
    require('@site/static/img/yktakaha4.jpg').default,
  ];
  return <Avatar src={src} />;
};
