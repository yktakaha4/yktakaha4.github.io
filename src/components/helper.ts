import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
export type CustomFields = {
  buildAt: string;
  commitHash: string;
  isDevelopment: boolean;
  isProduction: boolean;
  withEmbeddedContent: boolean;
};

export const getCustomFields = (): CustomFields => {
  const {
    siteConfig: { customFields },
  } = useDocusaurusContext();
  if (customFields) {
    return customFields as CustomFields;
  }
  throw new Error('customFields is not defined');
};
