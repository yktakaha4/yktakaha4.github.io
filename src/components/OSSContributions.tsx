import {FC, useMemo, useState} from "react";
import {OSSContribution} from "@/services/ossContributions";
import {createSearchRegexp, createSearchText, Search} from "@/components/ui/Search";
import dayjs from "dayjs";
import {Table, TableHeaders, TableRow} from "@/components/ui/Table";
import {getSliceIndex, Pager} from "@/components/ui/Pager";
import {getOSSContributionKindName, pagerPerPage, pagerSize} from "@/constants";

export type OSSContributionsProps = {
    contributions: Array<OSSContribution>
}

export const filterOSSContribution = (regexp: RegExp, contribution: OSSContribution) => {
    return regexp.test(createSearchText(
        contribution.title,
        dayjs(contribution.mergedAt).format('YYYY/M/D'),
        getOSSContributionKindName(contribution.kind),
        ...contribution.repository.languages,
        contribution.repository.name,
    ))
}

export const OSSContributions: FC<OSSContributionsProps> = ({ contributions }) => {
    const headers: TableHeaders = [{
        label: '日付',
        align: 'center',
        width: '10%',
    }, {
        label: 'タイトル',
        align: 'left',
        width: '40%',
    }, {
        label: 'リポジトリ',
        align: 'left',
        width: '40%',
    }, {
        label: 'タグ',
        align: 'left',
        width: '10%',
    }]

    const [query, setQuery] = useState('')
    const [page, setPage] = useState(1)

    const regexp = useMemo(() => createSearchRegexp(query), [query])

    const filtered = contributions.filter(contribution => filterOSSContribution(regexp, contribution))
    const rows = filtered.length > 0 ? filtered.map(({ mergedAt, title, url, repository, kind }): TableRow => {
        return [{
            type: 'date',
            value: mergedAt,
        }, {
            type: 'string',
            value: title,
            link: {
                type: 'external',
                href: url,
            },
        }, {
            type: 'string',
            value: repository.name,
            link: {
                type: 'external',
                href: repository.url,
            },
        }, {
            type: 'tags',
            values: [{
                icon: '📝',
                value: getOSSContributionKindName(kind),
                color: 'primary',
            }],
        }]
    }).slice(...getSliceIndex(page, pagerPerPage)) : []

    const handleChangeQuery = (newQuery: string) => {
        setQuery(newQuery)
        setPage(1)
    }
    const handleChangePage = (newPage: number) => {
        setPage(newPage)
    }

    return (<>
        <div style={{marginBottom: '0.2rem'}}>
            <Search
                query={query}
                totalCount={filtered.length}
                onChange={({ query }) => handleChangeQuery(query)}
                children={<Pager
                    page={page}
                    perPage={pagerPerPage}
                    size={pagerSize}
                    total={filtered.length}
                    onChange={({ page }) => handleChangePage(page)}
                />}
            />
        </div>
        <Table headers={headers} rows={rows} />
    </>)
}
