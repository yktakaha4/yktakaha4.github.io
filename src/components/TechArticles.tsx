import {FC, useMemo, useState} from "react";
import {createSearchRegexp, createSearchText, Search} from "@/components/ui/Search";
import {getSliceIndex, Pager} from "@/components/ui/Pager";
import {Table, TableHeaders, TableRow, TableRows} from "@/components/ui/Table";
import {pagerPerPage, pagerSize, TechArticlePublisher, getTechArticlePublisherName} from "@/constants";
import dayjs from "dayjs";

export type TechArticle = {
    publishedAt: Date
    title: string
    url: string
    likes: number
    publisher: TechArticlePublisher
    tags: Array<string>
}

export type TechArticlesProps = {
    articles: Array<TechArticle>
}

export const filterTechArticle = (regexp: RegExp, article: TechArticle) => {
    return regexp.test(createSearchText(
        article.title,
        ...article.tags,
        getTechArticlePublisherName(article.publisher),
        dayjs(article.publishedAt).format('YYYY/M/D'),
    ))
}

export const TechArticles: FC<TechArticlesProps> = ({ articles }) => {
    const headers: TableHeaders = [{
        label: 'å…¬é–‹æ—¥',
        align: 'center',
        width: '10%',
    }, {
        label: 'ã‚¿ã‚¤ãƒˆãƒ«',
        align: 'left',
        width: '70%',
    }, {
        label: 'ã‚¿ã‚°',
        align: 'left',
        width: '20%',
    }]

    const [query, setQuery] = useState('')
    const [page, setPage] = useState(1)

    const regexp = useMemo(() => createSearchRegexp(query), [query])

    const filteredArticles = articles.filter(article => filterTechArticle(regexp, article))
    const rows: TableRows = filteredArticles.map(({ publishedAt, title, url, likes, tags, publisher}): TableRow => {
        return [{
            type: 'date',
            value: publishedAt,
            format: 'YYYY/M/D',
        }, {
            type: 'string',
            value: title,
            link: {
                type: 'external',
                href: url,
            },
        }, {
            type: 'tags',
            values: [{
                icon: 'ðŸ“°',
                value: getTechArticlePublisherName(publisher),
                color: 'primary',
            }, {
                icon: 'â™¥',
                value: likes.toLocaleString(),
                color: 'danger',
            }, ...tags.map((tag) => ({ value: tag }))],
        }]
    }).slice(...getSliceIndex(page, pagerPerPage))

    const handleChangeQuery = (newQuery: string) => {
        setQuery(newQuery)
        setPage(1)
    }
    const handleChangePage = (newPage: number) => {
        setPage(newPage)
    }

    return (<>
        <div style={{marginBottom: '0.2rem'}}>
            <Search
                query={query}
                onChange={({ query }) => handleChangeQuery(query)}
                children={<Pager
                    page={page}
                    perPage={pagerPerPage}
                    size={pagerSize}
                    total={filteredArticles.length}
                    onChange={({ page }) => handleChangePage(page)}
                />}
            />
        </div>
        <Table headers={headers} rows={rows} />
    </>)
}
