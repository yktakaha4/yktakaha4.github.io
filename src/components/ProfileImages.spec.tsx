import {render} from "@testing-library/react";
import {ProfileImages} from "@/components/ProfileImages";

describe('ProfileImages', () => {
    describe('スナップショットテスト', () => {
        test('プロフィール画像が描画される', () => {
            const { container } = render(<ProfileImages />)
            expect(container).toMatchSnapshot()
        })
    })
})
