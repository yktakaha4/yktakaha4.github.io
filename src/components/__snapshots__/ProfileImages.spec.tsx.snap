// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileImages スナップショットテスト プロフィール画像が描画される 1`] = `
<div>
  <div
    class="avatar"
  >
    <a
      class="avatar__photo-link avatar__photo avatar__photo--lg"
      href="test-file-stub"
      target="_blank"
    >
      <img
        alt="アバター画像"
        src="test-file-stub"
        title="アバター画像"
      />
    </a>
    <a
      class="avatar__photo-link avatar__photo avatar__photo--lg"
      href="test-file-stub"
      target="_blank"
    >
      <img
        alt="アバター画像"
        src="test-file-stub"
        title="アバター画像"
      />
    </a>
    <div
      class="avatar__intro"
    >
      <div
        class="avatar__name"
      />
      <small
        class="avatar__subtitle"
      />
    </div>
  </div>
</div>
`;
